openmdao/components/tests/test_meta_model_structured_comp.py

```diff
@@ -1126,10 +1126,36 @@ def test_analysis_error_warning_msg(self):
         p.set_val('x', 0.75)
 
         msg = "Analysis Error: 'interp' <class MetaModelStructuredComp> " \
-              "Line 205 of file {}".format(inspect.getsourcefile(om.MetaModelStructuredComp))
+              "Line 210 of file {}".format(inspect.getsourcefile(om.MetaModelStructuredComp))
```

Comment:
Would be better to rewrite this check so it isn't so fragile.  Adding a line to MetaModelStructuredComp.py shouldn't break this test.

Reply:
The fault is the exception message actually, but to make the test more robust, I've added the capability to the `assert_warning` to optionally check if the warning contains the msg rather than check equality.